aten/src/ATen/native/SobolEngineOpsUtils.h

```diff
@@ -0,0 +1,1265 @@
+/// This file contains some tensor-agnostic operations to be used in the
+/// core functions of the `SobolEngine`
+#include <ATen/ATen.h>
+
+/// Function to return the minimum of number of bits to represent the integer `n`
+static inline int64_t bit_length(const int64_t n) {
+  int64_t nbits, nloc;
+  for (nloc = n, nbits = 0; nloc > 0; nloc /= 2, nbits++);
+  return nbits;
+}
+
+/// Function to get the position of the rightmost zero in the bit representation of an integer
+/// This value is the zero-indexed position
+static inline int64_t rightmost_zero(const int64_t n) {
+  int64_t z, i;
+  for (z = n, i = 0; z % 2 == 1; z /= 2, i++);
```

Comment:
Right shift (`>>`) might be faster than divide by 2

Reply:
Sure, I will incorporate this.

Reply:
No. Give some credit to compiler (it optimizes divide by 2), and keep your code readable.

Reply:
Is this function expected to return 64 when there are no zero bits in the input? If not, you should add a special handling for this corner case.

Reply:
I think returning 63 sounds reasonable.